sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -230,9 +231,23 @@ case class Union(children: Seq[LogicalPlan]) extends LogicalPlan {
   }
 
   // updating nullability to make all the children consistent
-  override def output: Seq[Attribute] =
-    children.map(_.output).transpose.map(attrs =>
-      attrs.head.withNullability(attrs.exists(_.nullable)))
+  override def output: Seq[Attribute] = {
+    children.map(_.output).transpose.map { attrs =>
+      val firstAttr = attrs.head
+      val outAttr = if (attrs.exists(_.isInstanceOf[UnresolvedAttribute])) {
+        firstAttr
```

Comment:
We need this case? I recall we don't call output when the plan is unresolved.

Reply:
I was not sure indeed. Let me remove it then. Thanks.